src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs

```diff
@@ -958,28 +958,25 @@ internal virtual void PrepareSession()
             {
                 WebSession.NoProxy = true;
             }
-            else
+            else if (Proxy is not null)
             {
-                if (Proxy is not null)
+                WebProxy webProxy = new(Proxy);
+
+                if (ProxyCredential is not null)
                 {
-                    WebProxy webProxy = new(Proxy);
-                    webProxy.BypassProxyOnLocal = false;
-                    if (ProxyCredential is not null)
-                    {
-                        webProxy.Credentials = ProxyCredential.GetNetworkCredential();
-                    }
-                    else
-                    {
-                        webProxy.UseDefaultCredentials =  ProxyUseDefaultCredentials;
-                    }
+                    webProxy.Credentials = ProxyCredential.GetNetworkCredential();
+                }
+                else
+                {
+                    webProxy.UseDefaultCredentials =  ProxyUseDefaultCredentials;
+                }
 
-                    // We don't want to update the WebSession unless the proxies are different
-                    // as that will require us to create a new HttpClientHandler and lose connection
-                    // persistence.
-                    if (!webProxy.Equals(WebSession.Proxy))
-                    {
-                        WebSession.Proxy = webProxy;
-                    }
+                // We don't want to update the WebSession unless the proxies are different
+                // as that will require us to create a new HttpClientHandler and lose connection
+                // persistence.
+                if (!webProxy.Equals(WebSession.Proxy))
```

Comment:
At simplest level, couldn't the comparison be something like this (excuse the possible namespace confusion, I've typed this directly into GitHub comment uncompiled/untested)?

```csharp
if (!string.IsNullOrEmpty(Proxy) && Proxy != (WebSession?.Proxy as System.Net.Http.Proxy)?.Address)
{
    if (WebSession.Proxy is System.Net.Http.Proxy wsProxy)
    {
        wsProxy.Address = Proxy;
    }
    else
    {
        // Proxy type is unexpected so replace with standard proxy (since the user supplied the Proxy parameter)
        WebSession.Proxy = new System.Net.Http.Proxy(Address, true);
    }
}
```

This may not pass the existing tests because reassigning Address from Proxy parameter when they're different won't cause the httpclient to be recreated, but it should work, and that test's expected result could be updated accordingly.

If you're also trying to deal with the `BypassProxyOnLocal` parameter (did you add that to the CmdLet because there's no way in current PowerShell to affect that setting), you'll need some extra logic for that too, but it can also be modified
directly on the Proxy class without needing to recreate it (provided the stored WebSession.Proxy is a System.Net.Http.Proxy object and not some custom Proxy class designed by the user).